src/mustache.cpp

```diff
@@ -187,7 +187,11 @@ void QtVariantContext::pop()
 int QtVariantContext::listCount(const QString& key) const
 {
 	const QVariant& item = value(key);
+#if QT_VERSION >= 0x060000
+	if (item.canConvert<QVariantList>() && item.typeId() != QMetaType::QString) {
```

Comment:
I think you can use `QVariant::userType` for cross-compatibility between Qt 5 and Qt 6.